src/database/remove.ts

```diff
@@ -0,0 +1,93 @@
+"use strict";
+
+import * as pathLib from "path";
+import { Queue } from "../queue";
+import FirebaseError = require("../error");
+import logger = require("../logger");
+import RemoveRemote from "./remove-remote";
+
+class DatabaseRemove {
+  public path: string;
+  public concurrency: number;
+  public retries: number;
+  public remote: any;
+  private jobQueue: Queue<string>;
+  private waitingPath: Map<string, number>;
+
+  /**
+   * Construct a new RTDB delete operation.
+   *
+   * @constructor
+   * @param {string} path path to delete.
+   * @param {string} options.instance the RTDB instance ID.
+   * @param {string} options.concurrency the number of concurrent chunk delete allowed
+   * @param {string} options.retires the number of retries for each chunk delete
+   */
+  constructor(path: string, options: any) {
```

Comment:
make an interface for `options` and use that! https://www.typescriptlang.org/docs/handbook/interfaces.html

```typescript
export interface DatabaseRemoveOptions {
  // RTBD instance ID.
  instance: string;
  // Number of concurrent chunk deletes to allow.
  concurrency: number;
  // Number of retries for each chunk delete.
  retries: number;
}
// ...
  constructor(path: string, options: DatabaseRemoveOptions) {
```

Even writing that I found that you have `string` instead of `number` in your jsdocs! 🙂   Also, `remote` is undocumented and used below.